Source/console-checkers/Source/GameState.cpp

```diff
@@ -0,0 +1,384 @@
+//---------------------------------------------------------------
+//
+// GameState.cpp
+//
+
+#include "Game.h"
+
+#include "GameSettings.h"
+#include "GameState.h"
+#include "GameTypes.h"
+#include "MoveDiscoveryEngine.h"
+#include "PlayerState.h"
+#include "Utility.h"
+
+#include <spdlog/spdlog.h>
+// Magic include needed for spdlog to log a custom type.
+// Must be included after spdlog.h. Thanks spdlog.
+#include <spdlog/fmt/ostr.h>
+#include <spdlog/fmt/ranges.h>
+
+#include <deque>
+#include <ranges>
+
+namespace Checkers {
+
+//===============================================================
+
+
+GameState::GameState(Game* game)
+	: m_gameBoard(GameplaySettings::s_defaultGameBoard.size())
+	, m_game(game)
+	, m_moveDiscoveryEngine(std::make_unique<MoveDiscoveryEngine>(this))
+{
+	// Creates both players.
+	m_playerStates.push_back(std::make_unique<PlayerState>(Identity::Red));
```

Comment:
If your functions are ...Red..() and ...Black...() you're already signing up for "there is only 2". If you truly planned on making this N instead of 2 your interface would reflect that.